### ISSUE TITLE
`navigationOptions` priority

### ISSUE THREAD

**Issue by jhhayashi** [react-community/react-navigation#482]
According to the docs, you can provide `navigationOptions` in three places:
- Static property on a screen
- Within a `Navigator`'s `RouteConfigs`
- Within a `Navigator`'s `StaticNavigatorConfig`

With the priority being:
> The navigationOptions from the screen will be merged key-by-key with the default options coming from the navigator. Whenever both the navigator and screen define the same option (e.g. header), the screen wins. 

And:
```javascript
StackNavigator({
  // For each screen that you can navigate to, create a new entry like this:
  Profile: {
    // Optional: Override the `navigationOptions` for the screen
    navigationOptions: {
      title: ({state}) => `${state.params.username}'s Profile'`,
    },
  },
});
```

i.e. `RouteConfigs > Screen Property > StaticNagivatorConfig`.

This seems a bit unintuitive, since the paradigm is that things defined most closely to anything bind most tightly. My intuition is that a title defined as part of the screen's properties should have the highest priority, since the user is actively viewing the screen.

I understand that the screen is technically embedded within the Navigator, and therefore a title defined within the Navigator itself would be the tightest binding and should therefore be the highest priority. But (especially in the case of a `StackNavigator`,) I tend to associate the title with the screen and not the `Navigator`.

In other words, my intuition is that `navigationOptions` defined as part of a screen should bind most tightly. Do other people have this intuition as well?

**Comment by satya164** [comment]
This allows you to have the same screen with some default config, and override some config when using the same screen in two different navigators. If you make the options in static options take higher priority it's no longer possible.

**Comment by nonameolsson** [comment]
@jhhayashi I agree! 

@satya164 That is true. But how could one override settings in the component?
